src/BinaryParsers/CommandLineHelper.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.BinaryParsers
+{
+    public static class CommandLineHelper
+    {
+        private static readonly char[] switchPrefix = new char[] { '-', '/' };
+
+        /// <summary>Possible status for commandline switches when searching the PDB (no default state is given).</summary>
+        public enum SwitchState
+        {
+            SwitchNotFound = 0,
+            SwitchEnabled = 1,
+            SwitchDisabled = -1
+        }
+
+        /// <summary>Order of precendence for the provided switches to clarify what to do in the presence of multiple disagreeing copies of the switch.</summary>
+        public enum OrderOfPrecedence
+        {
+            FirstWins = 0,
+            LastWins = 1
+        }
+
+        public static bool IsCommandLineOption(string candidate)
+        {
+            if (candidate.Length < 2)
+            {
+                return false;
+            }
+
+            char c = candidate[0];
+            return c == '/' || c == '-';
+        }
+
+        /// <summary>
+        /// Get the value of a compiler command-line options
+        /// </summary>
+        /// <param name="commandLine">Command line to search</param>
+        /// <param name="optionNames">Array of command line options to search for a value</param>
+        /// <param name="precedence">The precedence ruls for this set of options</param>
+        /// <param name="optionValue">string to recieve the value of the command line option</param>
+        /// <returns>true if one of the options is found, false if none are found</returns>
+        public static bool GetOptionValue(string commandLine, string[] optionNames, OrderOfPrecedence precedence, ref string optionValue)
+        {
+            bool optionFound = false;
+
+            if (optionNames != null && optionNames.Length > 0)
+            {
+                // array of strings for the options name without the preceding switchPrefix to make comparison easier
+                string[] optionsArray = new string[optionNames.Length]; ;
```

Comment:
>string[] [](http://example.com/codeflow?start=16&length=8)

use var in this case, since we know that u have a new string

Reply:
tried changing to var, VS show error icon and suggest to use IDE0008: use specific type instead of var.